### PR TITLE
ruler: fix the /api/v1/rules endpoint

### LINKED ISSUE #1189: rules api internal failed to assert type of rule
**Thanos, Prometheus and Golang version used**

```
thanos, version 0.4.0 (branch: HEAD, revision: a676095877e7a87576b0f7776fccd4b3f8b65188)
  build user:       root@3e7111579ad1
  build date:       20190504-09:31:11
  go version:       go1.11.9

start command 
/data/app/thanos-0.4.0.linux-amd64/thanos rule --cluster.disable --rule-file=rules/*.yml --query.sd-files=queryAPI.yml --http-address=0.0.0.0:10905 --grpc-address=0.0.0.0:10906 --alert.query-url=http://xxxx/thanos
```

**What happened**
```
/api/v1/rules

return:
{
  "status": "error",
  "ErrorType": "internal",
  "error": "failed to assert type of rule 'supervisor进程异常告警'"
}
```

My alert.yml

```
  rules:
  - alert: supervisor进程异常告警
    expr: |-
      node_supervisord_state{env="prd"}
        == 30 or node_supervisord_state{env="prd"} == 100 or node_supervisord_state{env="prd"}
        == 200 or node_supervisord_state{env="prd"} == 1000
    for: 1m
    labels:
      severity: critical
      uuid: 348436b0-2347-4f04-b61f-b6081543b192
      version: blitzcrank-alertmanager
    annotations:
      description: 服务名为{{ $labels.name}} {{$labels.ip}} Status:{{ $v:= printf "%v"
        $value }}{{if eq "20" $v }}RUNNING{{end}} {{if eq "30" $v }}BACKOFF{{end}}
        {{if eq "100" $v }}EXITED{{end}} {{if eq "200" $v }}FATAL{{end}} {{if eq "1000"
        $v }}UNKNOWN{{end}}
      job: cvm
      value: '{{$value}}'

```

**Anything else we need to know**

### PR DESCRIPTION
**Pull request by simonpasquier**
Closes #1189 

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Fixed the switch type in the `rules` handler because alerting rules are instances of `*rules.AlertingRule`, not `thanosrule.AlertingRule`. This is because the rules are returned from the Prometheus rule managers and have no notion of `PartialResponseStrategy`.

## Verification

The `/api/v1/rules` and `/api/v1/alerts` endpoints are now covered by the end-to-end tests.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -27,6 +27,7 @@ We use *breaking* word for marking changes that are not backward compatible (rel
 - [#1669](https://github.com/thanos-io/thanos/pull/1669) Fixed store sharding. Now it does not load excluded meta.jsons and load/fetch index-cache.json files.
 - [#1670](https://github.com/thanos-io/thanos/pull/1670) Fixed un-ordered blocks upload. Sidecar now uploads the oldest blocks first.
 - [#1568](https://github.com/thanos-io/thanos/pull/1709) Thanos Store now retains the first raw value of a chunk during downsampling to avoid losing some counter resets that occur on an aggregation boundary.
+- [#1773](https://github.com/thanos-io/thanos/pull/1773) Thanos Ruler: fixed the /api/v1/rules endpoint.
```
Comment (bwplotka) [inline CHANGELOG.md]: Can we tell users what exactly problem was fixed?

File: pkg/rule/api/v1_test.go [hunk 2]
```diff
@@ -96,43 +93,49 @@ func (m rulesRetrieverMock) RuleGroups() []thanosrule.Group {
...
+		ars = append(ars, thanosrule.AlertingRule{AlertingRule: ar})
+	}
+	return ars
+}
+
+func alertingRules() []*rules.AlertingRule {
 	expr1, err := promql.ParseExpr(`absent(test_metric3) != 1`)
 	if err != nil {
-		m.testing.Fatalf("unable to parse alert expression: %s", err)
+		panic(fmt.Sprintf("unable to parse alert expression: %s", err))
```
[the remark this answers is not in the document]
Comment (simonpasquier) [reply pkg/rule/api/v1_test.go]: Because it should never fail but I can revert if you prefer.

**Comment by s-urbaniak** [comment]
thanks @bwplotka for the review and @simonpasquier for the contribution! :tada:

**Pull request merged**